oletools/util/log_helper/_logger_adapter.py

```diff
@@ -0,0 +1,30 @@
+import logging
+from . import _root_logger_wrapper
+
+
+class OletoolsLoggerAdapter(logging.LoggerAdapter):
+    """
+    Adapter class for all loggers returned by the logging module.
+    """
+    _json_enabled = None
+
+    def print_str(self, message):
+        """
+        This function replaces normal print() calls so we can format them as JSON
+        when needed or just print them right away otherwise.
+        """
+        if self._json_enabled and self._json_enabled():
+            # Messages from this function should always be printed,
+            # so when using JSON we log using the same level that set
+            self.log(_root_logger_wrapper.level(), message)
+        else:
+            print(message)
+
+    def set_json_enabled_function(self, json_enabled):
```

Comment:
Since the control of whether JSON is enabled or not is done by the log_helper instance, this class receives a closure so it will always get the updated value of [`_json_enabled`](https://github.com/christian-intra2net/oletools/blob/7cb86312f0a9386926b56e9c8f5578090083edfe/oletools/util/log_helper/log_helper.py#L166).